tests/unit/auth/v4/queryAuthCheck.js

```diff
@@ -103,6 +106,32 @@ describe('v4 queryAuthCheck', () => {
         });
     });
 
+    it('should return error if host is not included as signed header', done => {
+        const alteredRequest = createAlteredRequest({ 'X-Amz-SignedHeaders':
+            'none' }, 'query', request, query);
+        const res = queryAuthCheck(alteredRequest, log, alteredRequest.query);
+        assert.deepStrictEqual(res.err, errors.AccessDenied);
+        done();
+    });
+
+    it('should return error if an x-amz header is not included as signed ' +
+        'header but is in request', done => {
+        const alteredRequest = createAlteredRequest({
+            'x-amz-acl': 'public' }, 'headers', request, headers);
+        const res = queryAuthCheck(alteredRequest, log, alteredRequest.query);
```

Comment:
The testing method is similar between the 3 functions, it could be worthwhile to encapsulate that.